File: swift/824_GoatLatin.swift
class Solution {
    let vowels: Set<Character> = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
    func toGoatLatin(_ S: String) -> String {
        var res: String = "", suffix: String = ""

        for currString in S.components(separatedBy: " ") {
            var curr = currString
            suffix += "a"
            if !vowels.contains(curr[curr.startIndex]) {
                let firstIdx = curr.index(curr.startIndex, offsetBy: 1)
                curr = "\(curr.substring(from: firstIdx))\(curr[curr.startIndex])"
            }
            res += curr + "ma" + suffix + " "
        }
        return String(res.dropLast())
    }
}

File: java/824_GoatLatin.java
class Solution {
    private static Set<Character> vowels =
            new HashSet<>() {
                {
                    add('a');
                    add('e');
                    add('i');
                    add('o');
                    add('u');
                    add('A');
                    add('E');
                    add('I');
                    add('O');
                    add('U');
                }
            };

    public String toGoatLatin(String S) {
        String suff = "";
        StringBuilder res = new StringBuilder();
        
        for (String curr : S.split(" ")) {
            suff += 'a';
            
            if (!vowels.contains(curr.charAt(0)))
                curr = curr.substring(1) + curr.charAt(0);
            
            res.append(curr).append("ma").append(suff).append(' ');
        }
        
        res.deleteCharAt(res.length() - 1);
        return res.toString();
    }
}
